qgispluginci/changelog.py

```diff
@@ -1,54 +1,165 @@
-"""Changelog parser.
+#! python3  # noqa E265
 
-Following nearly https://keepachangelog.com/en/1.0.0/
 """
+    Changelog parser. Following <https://keepachangelog.com/en/1.0.0/>.
+"""
+
+# ############################################################################
+# ########## Libraries #############
+# ##################################
 
+# standard library
+import logging
 import re
-from os.path import isfile
+from pathlib import Path
+from typing import NamedTuple, Union
+
+# ############################################################################
+# ########## Globals #############
+# ################################
+
+# see: https://regex101.com/r/ZpMhgM/1
+CHANGELOG_REGEXP = r"(?<=##)\s*\[*v?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)\]?(\(.*\))?(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?\]*\s-\s*([\d\-/]{10})(.*?)(?=##|\Z)"
+logger = logging.getLogger(__name__)
+
+# ############################################################################
+# ########## Classes #############
+# ################################
 
 
 class ChangelogParser:
-    @staticmethod
-    def has_changelog():
-        return isfile("CHANGELOG.md")
 
-    def __init__(self, regexp: str):
+    CHANGELOG_FILEPATH: Union[Path, None] = None
+
+    @classmethod
+    def has_changelog(cls, parent_folder: Union[Path, str] = Path(".")) -> bool:
+        """Check if a changelog file exists within the parent folder. If it does, \
```

Comment:
Do you need this trailing `\` at the end of all lines ?
Docstrings doesn't need this AFAIK :)

Reply:
Compare the output documentation with and without trailing slashes :wink:

Reply:
Well, as the number if "extraction" from the regexp (3 before) has changed, it's breaking anyway, no ?